python/ray/train/_internal/logging.py

```diff
@@ -0,0 +1,239 @@
+import logging
+import logging.config
+import os
+from enum import Enum
+from typing import Optional
+
+import yaml
+
+import ray
+from ray.train._internal.session import _get_session
+
+# JSON Encoding format for Ray Train structured logging
+DEFAULT_JSON_LOG_ENCODING_FORMAT = "JSON"
+# Env. variable to specify the encoding of the file logs when using the default config.
+LOG_ENCODING_ENV = "RAY_TRAIN_LOG_ENCODING"
+# Env. variable to specify the logging config path use defaults if not set
+LOG_CONFIG_PATH_ENV = "RAY_TRAIN_LOG_CONFIG_PATH"
+
+
+# Default logging configuration for Ray Train
+DEFAULT_LOG_CONFIG_JSON_STRING = {
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {
+        "ray": {
+            "format": "%(asctime)s\t%(levelname)s %(filename)s:%(lineno)s -- %(message)s"  # noqa: E501
+        },
+        "ray_json": {"class": "ray._private.ray_logging.formatters.JSONFormatter"},
+    },
+    "filters": {
+        "console_filter": {"()": "ray.train._internal.logging.HiddenRecordFilter"},
+        "core_context_filter": {
+            "()": "ray._private.ray_logging.filters.CoreContextFilter"
+        },
+        "train_context_filter": {
+            "()": "ray.train._internal.logging.TrainContextFilter"
+        },
+    },
+    "handlers": {
+        "console_json": {
+            "class": "logging.StreamHandler",
+            "formatter": "ray_json",
+            "filters": ["core_context_filter", "train_context_filter"],
+        },
+        "console_text": {
+            "class": "ray._private.log.PlainRayHandler",
+            "formatter": "ray",
+            "level": "INFO",
+            "filters": ["train_context_filter", "console_filter"],
+        },
+        "file": {
+            "class": "ray.train._internal.logging.SessionFileHandler",
+            "formatter": "ray_json",
+            "filename": "ray-train.log",
+            "filters": ["core_context_filter", "train_context_filter"],
+        },
+    },
+    "loggers": {
+        "ray.train": {
+            "level": "DEBUG",
+            "handlers": ["file", "console_text"],
+            "propagate": False,
+        },
+    },
+}
+
+
+class TrainLogKey(str, Enum):
+    RUN_ID = "run_id"
+    WORLD_SIZE = "world_size"
+    WORLD_RANK = "world_rank"
+    LOCAL_WORLD_SIZE = "local_world_size"
+    LOCAL_RANK = "local_rank"
+    NODE_RANK = "node_rank"
```

Comment:
I think we should only tag `world_rank`, `local_rank`, and `node_rank`. `world_size` / `local_world_size` is confusing to filter by.

Reply:
emm, good point. the size information is not useful for log searching / filtering.